src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,73 @@
+package duke.command;
+
+import duke.Parser;
+import duke.Ui;
+import duke.Storage;
+import duke.task.TaskList;
+import duke.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Represents command that is specific to the add command.
+ */
+public class AddCommand extends Command {
+
+    private String input;
+    private InputCommand inputCommand;
+
+    public AddCommand(String input, InputCommand inputCommand) {
+        this.exit = false;
+        this.input = input;
+        this.inputCommand = inputCommand;
+    }
+
+    /**
+     * Executes add command based on user input.
+     * @param taskList tasks of user.
+     * @param ui user interface object.
+     * @param storage Storage object to retrieve and store data from file.
+     * @throws DukeException If invalid input.
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+
+        switch (inputCommand) {
+        case TODO:
+            try {
+                taskList.addList(new Todo(Parser.getTodoDescription(input)));
+                ui.printAddedMessage(new Todo(Parser.getTodoDescription(input)), taskList.getListSize());
```

Comment:
I see that you do taskList.getListSize() several times. I think it is better to store it as a variable listSize to avoid calling getListSize() multiple times.